odoo/service/server.py

```diff
@@ -434,6 +438,45 @@ def target():
             t.start()
             _logger.debug("cron%d started!" % i)
 
+    def async_thread(self, n):
+        from odoo.addons.base.models.ir_async import ir_async
+        conn = odoo.sql_db.db_connect('postgres')
+
+        with conn.cursor() as cr:
+            pg_conn = cr._cnx
+            cr.execute("LISTEN odoo_async")
+            cr.commit()
+
+            notified_dbs = [
+                dbname
+                for dbname, registry
+                in odoo.modules.registry.Registry.registries.items()
+                if registry.ready
+            ]
+            while True:
+                time.sleep(n / 100 % .1)  # avoid thunderhead effect
+                for dbname in notified_dbs:
+                    _logger.debug("async%d processing db %s", n, dbname)
+                    threading.currentThread().start_time = time.time()
+                    ir_async._process_jobs(dbname)
+                threading.currentThread().start_time = None
+                _logger.debug("async%d awaiting signal", n)
+                select.select([pg_conn], [], [], SLEEP_INTERVAL)
+                pg_conn.poll()
```

Comment:
That doesn't seem correct: if the select() timed out, poll() will block until it has something to read or write. `select`'s result needs to be checked for the connection being ready.

Reply:
poll() doesn't wait for messages but you get a point, it is useless to poll if there is no new message

Reply:
Ah you're right I misread this bit:

>  If it returns POLL_OK then the connection has been established or the query results are available on the client.** Otherwise wait until the file descriptor returned by fileno() is ready to read or to write**

It probably tells the user to wait, but I kinda read it as poll doing the waiting.